x-pack/plugins/security_solution/cypress/e2e/detection_alerts/alert_table_hover_actions.cy.ts

```diff
@@ -0,0 +1,177 @@
+  ID_COLUMN_VALUES,
+  TOP_N_ALERT_HISTOGRAM,
+  TOP_N_CONTAINER_CLOSE_BTN,
+} from '../../screens/shared';
+import { HOVER_ACTIONS, TIMELINE_DATA_PROVIDERS_CONTAINER } from '../../screens/timeline';
+import { waitForAlerts, waitForTopNHistogramToLoad } from '../../tasks/alerts';
+import { createCustomRuleEnabled } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { waitForAlertsToPopulate } from '../../tasks/create_new_rule';
+import { login, visit } from '../../tasks/login';
+import { addsFieldsToTimeline } from '../../tasks/rule_details';
+import { closeTimelineUsingCloseButton } from '../../tasks/security_main';
+import { openActiveTimeline } from '../../tasks/timeline';
+import { ALERTS_URL } from '../../urls/navigation';
+
+describe('Alerts Table : Hover Actions', () => {
+  before(() => {
+    cleanKibana();
+    login();
+    createCustomRuleEnabled(getNewRule(), 'new custom rule');
+  });
+
+  context('Id Column', { testIsolation: false }, () => {
+    before(() => {
+      visit(ALERTS_URL);
+      waitForAlertsToPopulate();
+      const idColName = DATA_GRID_FIELDS.ID.fieldName;
+      addsFieldsToTimeline(idColName, [idColName]);
+      cy.get(GET_DATA_GRID_HEADER(DATA_GRID_FIELDS.ID.fieldName)).should('be.visible');
+    });
+    beforeEach(() => {
+      /*
+       *
+       * Since now alert table is different from the tables being used
+       * in events, we have to create different set of tests for hover actions
+       * for alerts table.
+       *
+       * These tests will be done on `_id` column since there can be difference
+       * in how `_id` is supplied to the security solution by trigger actions table
+       *
+       * */
+      waitForAlerts();
+      cy.get(ID_COLUMN_VALUES).first().trigger('mouseover', { timeout: 5000 });
+    });
+
+    it('Filter For', () => {
+      cy.get(ID_COLUMN_VALUES)
+        .eq(0)
+        .then(($el) => {
+          const idTextValue = $el.text();
+          cy.get(HOVER_ACTIONS.FILTER_FOR).should('exist');
+          cy.get(HOVER_ACTIONS.FILTER_FOR).trigger('click', { force: true });
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should('contain.text', `_id: ${idTextValue}`);
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM_DELETE).trigger('click');
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should('not.exist');
+        });
+    });
+
+    it('Filter Out', () => {
+      cy.get(ID_COLUMN_VALUES)
+        .eq(0)
+        .then(($el) => {
+          const idTextValue = $el.text();
+          cy.get(HOVER_ACTIONS.FILTER_OUT).should('exist');
+          cy.get(HOVER_ACTIONS.FILTER_OUT).trigger('click', { force: true });
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should('contain.text', `NOT _id: ${idTextValue}`);
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM_DELETE).trigger('click');
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should('not.exist');
+        });
+    });
+
+    it('Add To Timeline', () => {
```

Comment:
and this one got stuck in a weird loop checking the same thing over and over again, then finally failed

![Screenshot 2023-02-14 at 4 29 38 PM](https://user-images.githubusercontent.com/17276605/218877980-24776ceb-950b-4b2c-a664-dc32e365fc37.png)

Reply:
@PhilippeOberti , I will check why this is happening, thanks for checking it out.